uswds/uswds
# USWDS - Disabled styles: Improve consistency in high contrast mode

# Summary
Improved consistency of disabled styles in forced colors mode.

## Breaking change

This is not a breaking change.

## Related issue

Closes #5198

## Related pull requests

Changelog: https://github.com/uswds/uswds-site/pull/2102

## Preview link

Preview link: [Disabled form elements](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-disabled-high-contrast/?path=/story/patterns-forms--disabled-form-elements)

## Problem statement
PR https://github.com/uswds/uswds/pull/5063 introduces consistent disabled styles across all components. Viewing these components in high contrast mode exposes further inconsistency across components as well as missing icons & hover states.

## Solution
This PR adds styles for disabled styles in forced-color mode.

### Before:
![image](https://github.com/uswds/uswds/assets/93996430/db865fd9-6ba2-4f46-9865-710dc5a8c80e)

### After:
![image](https://github.com/uswds/uswds/assets/93996430/e807b543-c9e3-4d46-b2f8-777ea720ae62)


## Testing and review
### Turning on forced-colors mode in Chrome
1. Visit [Disabled form elements page](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-disabled-high-contrast/iframe.html?id=patterns-forms--disabled-form-elements&viewMode=story)
3. Open devTools
4. Press the three dot menu button in the top right corner of the devTools panel
5. Hover more tools and open rendering panel
6. Scroll down to `Emulate CSS media feature forced-colors`
7. Select `forced-colors: active`
8. Inspect Accordion component
9. In the rendering panel, find the `prefers-color-scheme` emulation and test high contrast in light and dark themes

### Tests
- Confirm that all form elements show a consistent and predictable disabled (and aria-disabled) style in forced color mode
    - Confirm that there is no change in the hover state
- Confirm no visual regressions in disabled form elements in standard mode
- Confirm code style
